lib/src/clojure_lsp/refactor/transform.clj

```diff
@@ -397,6 +397,46 @@
           (recur (first (expand-let loc false uri db)) current-edit)
           (some-> previous-edit vector))))))
 
+(defn new-defn-zloc [fn-name private? params body db]
+  (let [root (z/of-string
+               (cond
+                 (not private?)
+                 (format "(defn %s)" fn-name)
+
+                 (settings/get db [:use-metadata-for-privacy?] false)
+                 (format "(defn ^:private %s)" fn-name)
+
+                 :else
+                 (format "(defn- %s)" fn-name)))]
+    (reduce z/append-child*
+            (-> root
+                (z/append-child* (n/spaces 1))
+                (z/append-child* params))
+            body)))
+
+(defn prepend-preserving-comment
```

Comment:
Similarly, all three functions share logic for prepending the `defn` before the current form and before any comments on that form.

Reply:
This was done in two different ways before, only one of which preserved comment affiliation.